packages/core/rfcs/use-user-address-factory.md

```diff
@@ -0,0 +1,128 @@
+# UseUserAddressFactory with Commerce Tools composable use case
+
+## Motivation
+
+We'd like to separate a logic for the user addresses according to single responsibility principle. It's also more agnostic approach.
+
+## Factory interface
+
+```TS
+// factory schema
+export interface UseUserAddressFactory<ADDRESS, SEARCH_PARAMS> {
+  addresses: ComputedProperty<ADDRESS[]>;
+  totalAddresses: ComputedProperty<number>;
+  searchAddresses: (params?: SEARCH_PARAMS) => Promise<void>;
+  addShippingAddress: (address: ADDRESS) => Promise<void>;
+  addBillingAddress: (address: ADDRESS) => Promise<void>;
+  getShippingAddresses: () => ADDRESS[];
+  getBillingAddresses: () => ADDRESS[];
+  deleteAddress: (address: ADDRESS) => Promise<void>;
+  updateAddress: (address: ADDRESS) => Promise<void>;
+  loading: ComputedProperty<boolean>;
+}
+```
+
+## Factory params
+
+```TS
+// factory params schema
+export type UseUserAddressFactoryParams<ADDRESS, SEARCH_PARAMS> = {
+  searchAddresses: (params?: SEARCH_PARAMS) => Promise<ADDRESS[]>;
```

Comment:
according to our convention `loadAddresses` or `addressesSearch` if parametrized